force-app/main/default/classes/Gifts_TEST.cls

```diff
@@ -241,14 +338,57 @@ private class Gifts_TEST {
         return failReasonByElevateBatchId;
     }
 
+    public class DonationSelectorMock implements StubProvider {
+
+        public Integer numberOfQueriesRan = 0;
```

Comment:
This is actually asserting the number of times called for the getdonation or getdonations methods (not directly testing the # of queries called).  Should probably clarify that distinction in the tests.